### PR TITLE
refactor ExecuteScript body into Executor object

### PR DESCRIPTION
**Pull request by pattyshack**
Closes #???

## Description

prep work for parallelizing script preprocessing.  This is the second half of #1797

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@pattyshack Can you please follow and fill out the template that was provided (https://github.com/onflow/cadence/blob/master/.github/pull_request_template.md)?

File: runtime/runtime.go [hunk 1]
```diff
@@ -289,107 +289,17 @@ func (r *interpreterRuntime) SetDebugger(debugger *interpreter.Debugger) {
...
-		checkerOptions,
-		interpret,
-	)
-	if err != nil {
-		return nil, newError(err, context)
-	}
-
-	// Export before committing storage
+// TODO(patrick): return Executor instead of *interpreterScriptExecutor, and
+// update Executor / Runtime interfaces once PR #1797 merged.
```
[the remark this answers is not in the document]
Comment (pattyshack) [reply runtime/runtime.go:293]: I'm not sure what you mean by this comment.  afaik, github doesn't not support PR dependency.  but i'll make sure that #1797 is merged first, then update this PR accordingly

[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/runtime.go:293]: We're using [ZenHub](https://www.zenhub.com/). Once you installed the extension, you can add dependencies

File: runtime/runtime.go [hunk 2]
```diff
@@ -289,107 +289,17 @@ func (r *interpreterRuntime) SetDebugger(debugger *interpreter.Debugger) {
...
+func (r *interpreterRuntime) NewScriptExecutor(
+	script Script,
+	context Context) *interpreterScriptExecutor {
```
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/runtime.go:296]: This wasn't actually resolved

**Comment by pattyshack** [comment]
addressed comments + updated all interfaces

File: runtime/runtime.go [hunk 3]
```diff
@@ -98,6 +98,9 @@ type Executor interface {
 
 // Runtime is a runtime capable of executing Cadence.
 type Runtime interface {
+	// NewScriptExecutor returns an executor which executes the given script.
+	NewScriptExecutor(Script, Context) Executor
```
Comment (pattyshack) [inline runtime/runtime.go:102]: this is new

File: runtime/runtime.go [hunk 4]
```diff
@@ -355,107 +358,15 @@ func (r *interpreterRuntime) SetDebugger(debugger *interpreter.Debugger) {
...
-	// to storage will be actually persisted
-
-	err = r.commitStorage(storage, inter)
-	if err != nil {
-		return nil, newError(err, context)
-	}
+func (r *interpreterRuntime) NewScriptExecutor(
+	script Script,
+	context Context,
+) Executor {
```
Comment (pattyshack) [inline runtime/runtime.go:364]: the return value changed here

**Review by turbolent — APPROVED** [review]

**Pull request merged**